atlas/main_test.go

```diff
@@ -58,10 +59,31 @@ func TestMain(m *testing.M) {
 		log.Fatalf("failed to build server: %v", err)
 	}
 	log.Printf("runnning server")
-	if err := exec.CommandContext(ctx, fmt.Sprintf("%s/%s", basePath, output)).Start(); err != nil {
+	cmd := exec.CommandContext(ctx, fmt.Sprintf("%s/%s", basePath, output))
+	stderr, _ := cmd.StderrPipe()
+	cmdLog := bufio.NewScanner(stderr)
+	ready := make(chan struct{}, 1)
+	go func() {
+		first := true
+		for cmdLog.Scan() {
+			line := cmdLog.Text()
+			if first && strings.Contains(line, "serving") {
+				ready <- struct{}{}
+				first = false
+			}
+			log.Print(line)
+		}
+	}()
+	if err := cmd.Start(); err != nil {
 		log.Fatalf("failed to start server: %v", err)
 	}
+	go func() {
+		if err := cmd.Wait(); err != nil {
+			log.Fatal()
+		}
+	}()
 	log.Print("wait for servers to load up")
+	<-ready
```

Comment:
This adds two features to improve the tests, since they have been flakey. First, it copies the logs from the server into the test logs, so we can see any problem. Also, it waits until the server reports that it is serving before running any other tests, so we don't get transient readiness errors from the other tests.